internal/pkg/filesystem_impl/mount.go

```diff
@@ -69,9 +66,6 @@ func mount(fsType FSType, sessionName datamodel.SessionName, isMultiJob bool, in
 			if err := mkdir(attachHost, privateDir); err != nil {
 				return err
 			}
-			if err := fixUpOwnership(attachHost, owner, group, privateDir); err != nil {
```

Comment:
I am planning to move the creation of these directories so it only happens on the primary brick host during create. However in that version it means we still need the mkdir here, instead the private directoy and global directory will get the permissions set spearately.

Now that is a thing we should do with install.

So maybe we should change this more, i.e.:
* add create of the /private and /global directories before this larget mount loop.
* only need to do the install on one host really, so maybe just pick the first host and do it there?

I might try do a bit of this rework myself it help kick this off...

Reply:
Sorry I missed this comment before closing and re-creating the pull request. Would it make sense for you to pull in my branch and continue working on it? I'll leave it with you, thanks.